src/shared/TestInfrastructure/TestUtils.cs

```diff
@@ -87,5 +89,37 @@ public static string GetUuid(int length = -1)
 
             return uuid.Substring(0, length);
         }
+
+        public static async Task<string> RunCommandAsync(string filePath, string arguments, string workingDirectory = null)
+        {
+            using var process = new Process
+            {
+                StartInfo = new ProcessStartInfo
+                {
+                    FileName = filePath,
+                    Arguments = arguments,
+                    RedirectStandardOutput = true,
+                    RedirectStandardError = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                    WorkingDirectory = workingDirectory ?? Environment.CurrentDirectory
+                }
+            };
+
+            process.Start();
+
+            string output = await process.StandardOutput.ReadToEndAsync();
+            string error = await process.StandardError.ReadToEndAsync();
+
+            await process.WaitForExitAsync();
+
+            if (process.ExitCode != 0)
+            {
+                throw new InvalidOperationException(
+                    $"Command `{filePath} {arguments}` failed with exit code {process.ExitCode}. Error: {error}");
```

Comment:
Depending on how general purpose you intend this to be, it may be good to include stdout in this exception as well.

Reply:
Good idea. Added.